cmd/autoscaler/main.go

```diff
@@ -159,14 +165,74 @@ func main() {
 		logger.Fatalw("Failed to start informers", zap.Error(err))
 	}
 
-	go controller.StartAll(ctx, controllers...)
+	// Needs to be called after informaers have started.
+	ss, err := statefulsetLister.StatefulSets(ns).Get(statefulSetName)
+	if err != nil {
+		logger.Fatalw("Failed to get autoscaler StatefulSet", zap.Error(err))
+	}
 
-	go func() {
+	acceptSm := func(sm asmetrics.StatMessage) {
```

Comment:
what's `Sm` here? I'd expand the name a bit. Same `Sms` below.

Reply:
done